File: android/app/src/main/java/com/retroplay/GameView.java
package com.retroplay;

import android.content.Context;
import android.graphics.Color;
import android.os.Build;
import android.os.Environment;
import android.util.DisplayMetrics;
import android.view.Display;
import android.widget.FrameLayout;

import androidx.lifecycle.Lifecycle;

import com.swordfish.libretrodroid.GLRetroView;
import com.swordfish.libretrodroid.GLRetroViewData;

public class GameView extends FrameLayout {
    GLRetroViewData data;
    public GameView(Context context, Lifecycle lifecycle,String romName,String coreName){
        super(context);

        data = new GLRetroViewData(context);
        data.setCoreFilePath("libmgba_libretro_android.so");
        data.setGameFilePath(this.getContext().getExternalFilesDir(Environment.DIRECTORY_DOCUMENTS)+ "/example.gba");
        data.setGameFileBytes(null);


        data.setRumbleEventsEnabled(false);
        data.setPreferLowLatencyAudio(true);
       GLRetroView retroView = new GLRetroView(context,data);
        DisplayMetrics metrics = new DisplayMetrics();
       context.getDisplay().getMetrics(metrics);
        retroView.setMinimumHeight(metrics.heightPixels);
        retroView.setMinimumWidth(metrics.widthPixels);

//        this.layout(0,0,500,500);
       // this.setBackgroundColor(Color.GREEN);
       lifecycle.addObserver(retroView);
       this.addView(retroView);

    }
}

File: android/app/src/main/java/com/retroplay/GameView.kt
package com.retroplay

import android.content.Context
import android.os.Environment

import android.widget.FrameLayout
import com.swordfish.libretrodroid.GLRetroViewData
import com.swordfish.libretrodroid.GLRetroView
import com.swordfish.libretrodroid.AspectRatioGLSurfaceView
import com.retroplay.GameView
import com.swordfish.radialgamepad.library.RadialGamePad
import android.view.LayoutInflater
import com.retroplay.VirtualGamePadConfigs
import android.view.View.OnTouchListener
import androidx.lifecycle.Lifecycle
import com.facebook.react.ReactPackage
import com.facebook.react.bridge.ReactApplicationContext
import com.retroplay.RNGameManager

class GameView(context: Context?, lifecycle: Lifecycle, romName: String?, coreName: String?) :
    FrameLayout(
        context!!
    ) {
    var data: GLRetroViewData

    init {
        data = GLRetroViewData(context!!)
        data.coreFilePath = "libmgba_libretro_android.so"
        data.gameFilePath =
            context.getExternalFilesDir(Environment.DIRECTORY_DOCUMENTS).toString() + "/example.gba"
        data.gameFileBytes = null
        data.rumbleEventsEnabled = true
        data.preferLowLatencyAudio = true
        val retroView = GLRetroView(context, data)
        retroView.id = R.id.gl_retro_view
        retroView.setResizeMode(AspectRatioGLSurfaceView.RESIZE_MODE_FILL)
        //        DisplayMetrics metrics = new DisplayMetrics();
//       context.getDisplay().getMetrics(metrics);
//        retroView.setMinimumHeight(metrics.heightPixels);
//        retroView.setMinimumWidth(metrics.widthPixels);

//        this.layout(0,0,500,500);
        // this.setBackgroundColor(Color.GREEN);
        lifecycle.addObserver(retroView)
        this.addView(retroView)
    }
}